Enable -Wunsafe-buffer-usage in WebKit.framework

geoffreygaren opened the pull request:
#### 8012a2ef24887940831a4ddc777d5d757f523c99
<pre>
Enable -Wunsafe-buffer-usage in WebKit.framework
<a href="https://bugs.webkit.org/show_bug.cgi?id=281252">https://bugs.webkit.org/show_bug.cgi?id=281252</a>
<a href="https://rdar.apple.com/137710184">rdar://137710184</a>

Reviewed by Ryosuke Niwa.

This patch enables -Wunsafe-buffer-usage by default in WebKit.framework.

This is our bounds safety strategy for Safer C++ in WebKit.

For the time being, files that do not conform to safe buffer usage opt out,
using a new macro. The next step is to burn down the opt out list.

In cases where a header triggers a warning, we either opt out in a chokepoint
header that includes that header, or opt out the header directly, whichever is
more expedient.

I also fixed exactly one unsafe case, to try it out, and also to work around
a header #include path issue I couldn&apos;t figure out.

* Source/JavaScriptCore/API/APICast.h:
* Source/JavaScriptCore/JavaScriptCorePrefix.h:
* Source/JavaScriptCore/heap/Strong.h:
* Source/JavaScriptCore/heap/WeakInlines.h:
* Source/JavaScriptCore/interpreter/CallFrame.h:
* Source/JavaScriptCore/runtime/ArrayBuffer.h:
* Source/JavaScriptCore/runtime/ExceptionScope.h:
* Source/JavaScriptCore/runtime/GenericTypedArrayViewInlines.h:
* Source/JavaScriptCore/runtime/JSArrayBufferViewInlines.h:
* Source/JavaScriptCore/runtime/JSBigInt.h:
* Source/JavaScriptCore/runtime/JSCJSValueInlines.h:
* Source/JavaScriptCore/runtime/JSCellInlines.h:
* Source/JavaScriptCore/runtime/JSObject.h:
* Source/JavaScriptCore/runtime/Options.h:
* Source/JavaScriptCore/runtime/TypedArrays.h:
* Source/JavaScriptCore/runtime/VM.h:
* Source/JavaScriptCore/wasm/WasmModule.h:
* Source/WTF/wtf/ASCIICType.h:
* Source/WTF/wtf/Atomics.h:
* Source/WTF/wtf/BitSet.h:
* Source/WTF/wtf/BitVector.h:
* Source/WTF/wtf/BloomFilter.h:
* Source/WTF/wtf/Brigand.h:
* Source/WTF/wtf/Compiler.h:
* Source/WTF/wtf/CrossThreadCopier.h:
* Source/WTF/wtf/Deque.h:
* Source/WTF/wtf/FileSystem.h:
* Source/WTF/wtf/FixedVector.h:
* Source/WTF/wtf/HashMap.h:
* Source/WTF/wtf/HashSet.h:
* Source/WTF/wtf/HexNumber.h:
* Source/WTF/wtf/IteratorAdaptors.h:
* Source/WTF/wtf/Locker.h:
* Source/WTF/wtf/ObjectIdentifier.h:
* Source/WTF/wtf/PageBlock.h:
* Source/WTF/wtf/RefCounted.h:
* Source/WTF/wtf/RobinHoodHashTable.h:
* Source/WTF/wtf/SIMDHelpers.h:
* Source/WTF/wtf/SortedArrayMap.h:
* Source/WTF/wtf/StackTrace.h:
* Source/WTF/wtf/StdLibExtras.h:
* Source/WTF/wtf/TZoneMallocInlines.h:
* Source/WTF/wtf/Threading.h:
* Source/WTF/wtf/TrailingArray.h:
* Source/WTF/wtf/UUID.h:
* Source/WTF/wtf/Vector.h:
* Source/WTF/wtf/WTFConfig.h:
* Source/WTF/wtf/cf/VectorCF.h:
* Source/WTF/wtf/cocoa/SpanCocoa.h:
(WTF::span):
* Source/WTF/wtf/text/ASCIIFastPath.h:
* Source/WTF/wtf/text/ASCIILiteral.h:
* Source/WTF/wtf/text/CString.h:
* Source/WTF/wtf/text/MakeString.h:
* Source/WTF/wtf/text/StringBuilder.h:
* Source/WTF/wtf/text/StringCommon.h:
* Source/WTF/wtf/text/StringConcatenate.h:
* Source/WTF/wtf/text/StringConcatenateNumbers.h:
* Source/WTF/wtf/text/StringHasherInlines.h:
* Source/WTF/wtf/text/StringImpl.h:
* Source/WTF/wtf/text/StringView.h:
* Source/WTF/wtf/text/TextBreakIterator.h:
* Source/WTF/wtf/text/WTFString.h:
* Source/WTF/wtf/text/icu/TextBreakIteratorICU.h:
* Source/WebCore/Scripts/SettingsTemplates/Settings.h.erb:
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/bindings/IDLTypes.h:
* Source/WebCore/bindings/js/BufferSource.h:
* Source/WebCore/bindings/js/DOMWrapperWorld.h:
* Source/WebCore/bindings/js/JSDOMConvertBase.h:
* Source/WebCore/bindings/js/JSDOMGlobalObject.h:
* Source/WebCore/bindings/js/JSDOMWrapper.h:
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/CSSSelectorList.h:
* Source/WebCore/css/parser/CSSParserTokenRange.h:
* Source/WebCore/dom/ElementData.h:
* Source/WebCore/dom/SpaceSplitString.h:
* Source/WebCore/dom/messageports/MessagePortChannel.h:
* Source/WebCore/page/StructuredSerializeOptions.h:
* Source/WebCore/platform/ProcessQualified.h:
* Source/WebCore/platform/SharedBuffer.h:
* Source/WebCore/platform/SharedMemory.h:
* Source/WebCore/platform/audio/AudioChannel.h:
* Source/WebCore/platform/graphics/FontTaggedSettings.h:
* Source/WebCore/platform/graphics/FourCC.h:
* Source/WebCore/platform/graphics/GlyphBuffer.h:
* Source/WebCore/platform/graphics/GlyphPage.h:
* Source/WebCore/platform/graphics/WidthCache.h:
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
* Source/WebCore/platform/network/HTTPHeaderMap.h:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/GPUProcess/EntryPoint/Cocoa/XPCService/GPUServiceEntryPoint.mm:
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.h:
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.cpp:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp:
* Source/WebKit/GPUProcess/webrtc/LibWebRTCCodecsProxy.mm:
* Source/WebKit/NetworkProcess/Downloads/DownloadMonitor.cpp:
* Source/WebKit/NetworkProcess/Notifications/Cocoa/WebPushDaemonConnectionCocoa.mm:
* Source/WebKit/NetworkProcess/PrivateClickMeasurement/PrivateClickMeasurementDatabase.cpp:
* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerToContextConnection.cpp:
* Source/WebKit/NetworkProcess/cache/NetworkCacheDataCocoa.mm:
* Source/WebKit/NetworkProcess/storage/BackgroundFetchStoreManager.cpp:
* Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
* Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.cpp:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportReceiveStreamCocoa.mm:
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm:
* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
* Source/WebKit/Platform/IPC/DaemonCoders.h:
* Source/WebKit/Platform/IPC/DaemonDecoder.cpp:
* Source/WebKit/Platform/IPC/Decoder.cpp:
* Source/WebKit/Platform/IPC/Encoder.cpp:
* Source/WebKit/Platform/IPC/Encoder.h:
* Source/WebKit/Platform/IPC/JSIPCBinding.h:
* Source/WebKit/Platform/IPC/StreamConnectionBuffer.h:
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm:
* Source/WebKit/Platform/IPC/cocoa/DaemonConnectionCocoa.mm:
* Source/WebKit/Scripts/webkit/messages.py:
(generate_message_names_header):
* Source/WebKit/Scripts/webkit/tests/MessageNames.h:
* Source/WebKit/Shared/API/APIData.h:
* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
* Source/WebKit/Shared/API/Cocoa/_WKRemoteObjectInterface.mm:
* Source/WebKit/Shared/API/c/WKArray.cpp:
* Source/WebKit/Shared/API/c/WKData.cpp:
* Source/WebKit/Shared/API/c/WKDictionary.cpp:
* Source/WebKit/Shared/API/c/WKString.cpp:
* Source/WebKit/Shared/API/c/WKURLRequest.cpp:
* Source/WebKit/Shared/Cocoa/CoreIPCCFCharacterSet.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDateComponents.mm:
* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.h:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/Daemon/PCMDaemonEntryPoint.mm:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.h:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceMain.mm:
* Source/WebKit/Shared/RTCNetwork.h:
* Source/WebKit/Shared/RTCWebKitEncodedFrameInfo.h:
* Source/WebKit/Shared/SharedStringHashStore.cpp:
* Source/WebKit/Shared/SharedStringHashTableReadOnly.cpp:
* Source/WebKit/Shared/cf/CoreIPCSecAccessControl.h:
* Source/WebKit/Shared/cf/CoreIPCSecCertificate.h:
* Source/WebKit/Shared/cf/CoreIPCSecKeychainItem.h:
* Source/WebKit/Shared/cf/CoreIPCSecTrust.h:
* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
* Source/WebKit/Shared/mac/WebMemorySampler.mac.mm:
* Source/WebKit/UIProcess/API/APIContentRuleListStore.cpp:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.cpp:
* Source/WebKit/UIProcess/UserMediaPermissionRequestManagerProxy.cpp:
* Source/WebKit/UIProcess/UserMediaPermissionRequestProxy.cpp:
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/HidConnection.mm:
* Source/WebKit/UIProcess/WebAuthentication/Mock/MockLocalConnection.mm:
* Source/WebKit/UIProcess/WebAuthentication/fido/CtapHidDriver.cpp:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
* Source/WebKit/WebProcess/Extensions/Bindings/Scripts/CodeGeneratorExtensions.pm:
(_generateImplementationFile):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteComputePassEncoderProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/cocoa/RemoteGraphicsContextGLProxyCocoa.mm:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioDestinationProxy.cpp:
* Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
* Source/WebKit/WebProcess/GPU/webrtc/SharedVideoFrame.cpp:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCDnsResolverFactory.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCSocket.cpp:
* Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.mm:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
* Source/WebKit/WebProcess/Plugins/PDF/PDFScriptEvaluator.mm:
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
* Source/WebKit/WebProcess/WebPage/MomentumEventDispatcher.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/webpushd/WebPushDaemon.mm:
* Source/WebKit/webpushd/webpushtool/WebPushToolConnection.mm:
* Source/WebKitLegacy/mac/WebKitPrefix.h:

Canonical link: <a href="https://commits.webkit.org/285046@main">https://commits.webkit.org/285046@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4d20b73796f5fa115cdff4f2ffaf6eefb68ef0ae

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71380 "15 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50793 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24153 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/75490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/22585 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/73495 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58593 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22404 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/75490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/22585 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74446 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46117 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/61500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/75490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/70888 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/42783 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/18964 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/20926 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/64505 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/64682 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19330 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/77210 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/70629 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15614 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/18509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/77210 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15657 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61539 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/77210 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12251 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/5882 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/92415 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10939 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46593 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/1372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/20380 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47664 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48947 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/47406 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

rniwa reviewed (approved):
Do we need to test perf? or does this feature only affect compile time work and not runtime?

geoffreygaren committed:
Enable -Wunsafe-buffer-usage in WebKit.framework
https://bugs.webkit.org/show_bug.cgi?id=281252
rdar://137710184

Reviewed by Ryosuke Niwa.

This patch enables -Wunsafe-buffer-usage by default in WebKit.framework.

This is our bounds safety strategy for Safer C++ in WebKit.

For the time being, files that do not conform to safe buffer usage opt out,
using a new macro. The next step is to burn down the opt out list.

In cases where a header triggers a warning, we either opt out in a chokepoint
header that includes that header, or opt out the header directly, whichever is
more expedient.

I also fixed exactly one unsafe case, to try it out, and also to work around
a header #include path issue I couldn't figure out.

* Source/JavaScriptCore/API/APICast.h:
* Source/JavaScriptCore/JavaScriptCorePrefix.h:
* Source/JavaScriptCore/heap/Strong.h:
* Source/JavaScriptCore/heap/WeakInlines.h:
* Source/JavaScriptCore/interpreter/CallFrame.h:
* Source/JavaScriptCore/runtime/ArrayBuffer.h:
* Source/JavaScriptCore/runtime/ExceptionScope.h:
* Source/JavaScriptCore/runtime/GenericTypedArrayViewInlines.h:
* Source/JavaScriptCore/runtime/JSArrayBufferViewInlines.h:
* Source/JavaScriptCore/runtime/JSBigInt.h:
* Source/JavaScriptCore/runtime/JSCJSValueInlines.h:
* Source/JavaScriptCore/runtime/JSCellInlines.h:
* Source/JavaScriptCore/runtime/JSObject.h:
* Source/JavaScriptCore/runtime/Options.h:
* Source/JavaScriptCore/runtime/TypedArrays.h:
* Source/JavaScriptCore/runtime/VM.h:
* Source/JavaScriptCore/wasm/WasmModule.h:
* Source/WTF/wtf/ASCIICType.h:
* Source/WTF/wtf/Atomics.h:
* Source/WTF/wtf/BitSet.h:
* Source/WTF/wtf/BitVector.h:
* Source/WTF/wtf/BloomFilter.h:
* Source/WTF/wtf/Brigand.h:
* Source/WTF/wtf/Compiler.h:
* Source/WTF/wtf/CrossThreadCopier.h:
* Source/WTF/wtf/Deque.h:
* Source/WTF/wtf/FileSystem.h:
* Source/WTF/wtf/FixedVector.h:
* Source/WTF/wtf/HashMap.h:
* Source/WTF/wtf/HashSet.h:
* Source/WTF/wtf/HexNumber.h:
* Source/WTF/wtf/IteratorAdaptors.h:
* Source/WTF/wtf/Locker.h:
* Source/WTF/wtf/ObjectIdentifier.h:
* Source/WTF/wtf/PageBlock.h:
* Source/WTF/wtf/RefCounted.h:
* Source/WTF/wtf/RobinHoodHashTable.h:
* Source/WTF/wtf/SIMDHelpers.h:
* Source/WTF/wtf/SortedArrayMap.h:
* Source/WTF/wtf/StackTrace.h:
* Source/WTF/wtf/StdLibExtras.h:
* Source/WTF/wtf/TZoneMallocInlines.h:
* Source/WTF/wtf/Threading.h:
* Source/WTF/wtf/TrailingArray.h:
* Source/WTF/wtf/UUID.h:
* Source/WTF/wtf/Vector.h:
* Source/WTF/wtf/WTFConfig.h:
* Source/WTF/wtf/cf/VectorCF.h:
* Source/WTF/wtf/cocoa/SpanCocoa.h:
(WTF::span):
* Source/WTF/wtf/text/ASCIIFastPath.h:
* Source/WTF/wtf/text/ASCIILiteral.h:
* Source/WTF/wtf/text/CString.h:
* Source/WTF/wtf/text/MakeString.h:
* Source/WTF/wtf/text/StringBuilder.h:
* Source/WTF/wtf/text/StringCommon.h:
* Source/WTF/wtf/text/StringConcatenate.h:
* Source/WTF/wtf/text/StringConcatenateNumbers.h:
* Source/WTF/wtf/text/StringHasherInlines.h:
* Source/WTF/wtf/text/StringImpl.h:
* Source/WTF/wtf/text/StringView.h:
* Source/WTF/wtf/text/TextBreakIterator.h:
* Source/WTF/wtf/text/WTFString.h:
* Source/WTF/wtf/text/icu/TextBreakIteratorICU.h:
* Source/WebCore/Scripts/SettingsTemplates/Settings.h.erb:
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/bindings/IDLTypes.h:
* Source/WebCore/bindings/js/BufferSource.h:
* Source/WebCore/bindings/js/DOMWrapperWorld.h:
* Source/WebCore/bindings/js/JSDOMConvertBase.h:
* Source/WebCore/bindings/js/JSDOMGlobalObject.h:
* Source/WebCore/bindings/js/JSDOMWrapper.h:
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/CSSSelectorList.h:
* Source/WebCore/css/parser/CSSParserTokenRange.h:
* Source/WebCore/dom/ElementData.h:
* Source/WebCore/dom/SpaceSplitString.h:
* Source/WebCore/dom/messageports/MessagePortChannel.h:
* Source/WebCore/page/StructuredSerializeOptions.h:
* Source/WebCore/platform/ProcessQualified.h:
* Source/WebCore/platform/SharedBuffer.h:
* Source/WebCore/platform/SharedMemory.h:
* Source/WebCore/platform/audio/AudioChannel.h:
* Source/WebCore/platform/graphics/FontTaggedSettings.h:
* Source/WebCore/platform/graphics/FourCC.h:
* Source/WebCore/platform/graphics/GlyphBuffer.h:
* Source/WebCore/platform/graphics/GlyphPage.h:
* Source/WebCore/platform/graphics/WidthCache.h:
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
* Source/WebCore/platform/network/HTTPHeaderMap.h:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/GPUProcess/EntryPoint/Cocoa/XPCService/GPUServiceEntryPoint.mm:
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.h:
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.cpp:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp:
* Source/WebKit/GPUProcess/webrtc/LibWebRTCCodecsProxy.mm:
* Source/WebKit/NetworkProcess/Downloads/DownloadMonitor.cpp:
* Source/WebKit/NetworkProcess/Notifications/Cocoa/WebPushDaemonConnectionCocoa.mm:
* Source/WebKit/NetworkProcess/PrivateClickMeasurement/PrivateClickMeasurementDatabase.cpp:
* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerToContextConnection.cpp:
* Source/WebKit/NetworkProcess/cache/NetworkCacheDataCocoa.mm:
* Source/WebKit/NetworkProcess/storage/BackgroundFetchStoreManager.cpp:
* Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
* Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.cpp:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportReceiveStreamCocoa.mm:
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm:
* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
* Source/WebKit/Platform/IPC/DaemonCoders.h:
* Source/WebKit/Platform/IPC/DaemonDecoder.cpp:
* Source/WebKit/Platform/IPC/Decoder.cpp:
* Source/WebKit/Platform/IPC/Encoder.cpp:
* Source/WebKit/Platform/IPC/Encoder.h:
* Source/WebKit/Platform/IPC/JSIPCBinding.h:
* Source/WebKit/Platform/IPC/StreamConnectionBuffer.h:
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm:
* Source/WebKit/Platform/IPC/cocoa/DaemonConnectionCocoa.mm:
* Source/WebKit/Scripts/webkit/messages.py:
(generate_message_names_header):
* Source/WebKit/Scripts/webkit/tests/MessageNames.h:
* Source/WebKit/Shared/API/APIData.h:
* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
* Source/WebKit/Shared/API/Cocoa/_WKRemoteObjectInterface.mm:
* Source/WebKit/Shared/API/c/WKArray.cpp:
* Source/WebKit/Shared/API/c/WKData.cpp:
* Source/WebKit/Shared/API/c/WKDictionary.cpp:
* Source/WebKit/Shared/API/c/WKString.cpp:
* Source/WebKit/Shared/API/c/WKURLRequest.cpp:
* Source/WebKit/Shared/Cocoa/CoreIPCCFCharacterSet.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDateComponents.mm:
* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.h:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/Daemon/PCMDaemonEntryPoint.mm:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.h:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceMain.mm:
* Source/WebKit/Shared/RTCNetwork.h:
* Source/WebKit/Shared/RTCWebKitEncodedFrameInfo.h:
* Source/WebKit/Shared/SharedStringHashStore.cpp:
* Source/WebKit/Shared/SharedStringHashTableReadOnly.cpp:
* Source/WebKit/Shared/cf/CoreIPCSecAccessControl.h:
* Source/WebKit/Shared/cf/CoreIPCSecCertificate.h:
* Source/WebKit/Shared/cf/CoreIPCSecKeychainItem.h:
* Source/WebKit/Shared/cf/CoreIPCSecTrust.h:
* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
* Source/WebKit/Shared/mac/WebMemorySampler.mac.mm:
* Source/WebKit/UIProcess/API/APIContentRuleListStore.cpp:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.cpp:
* Source/WebKit/UIProcess/UserMediaPermissionRequestManagerProxy.cpp:
* Source/WebKit/UIProcess/UserMediaPermissionRequestProxy.cpp:
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/HidConnection.mm:
* Source/WebKit/UIProcess/WebAuthentication/Mock/MockLocalConnection.mm:
* Source/WebKit/UIProcess/WebAuthentication/fido/CtapHidDriver.cpp:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
* Source/WebKit/WebProcess/Extensions/Bindings/Scripts/CodeGeneratorExtensions.pm:
(_generateImplementationFile):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteComputePassEncoderProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/cocoa/RemoteGraphicsContextGLProxyCocoa.mm:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioDestinationProxy.cpp:
* Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
* Source/WebKit/WebProcess/GPU/webrtc/SharedVideoFrame.cpp:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCDnsResolverFactory.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCSocket.cpp:
* Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.mm:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
* Source/WebKit/WebProcess/Plugins/PDF/PDFScriptEvaluator.mm:
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
* Source/WebKit/WebProcess/WebPage/MomentumEventDispatcher.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/webpushd/WebPushDaemon.mm:
* Source/WebKit/webpushd/webpushtool/WebPushToolConnection.mm:
* Source/WebKitLegacy/mac/WebKitPrefix.h:

Canonical link: https://commits.webkit.org/285046@main
